app/js/PersistorHelper.js

```diff
@@ -23,8 +24,11 @@ class ObserverStream extends Stream.Transform {
       this.hash = crypto.createHash(options.hash)
     }
     if (options.metric) {
-      const onEnd = () => {
+      const onEnd = err => {
         metrics.count(options.metric, this.bytes)
+        if (err) {
+          this.destroy(err)
+        }
```

Comment:
The `.destroy(err)` invocation will emit another `'error'` event, which is not allowed per the docs.

> https://nodejs.org/api/stream.html#stream_writable_destroy_error
> 
> `writable.destroy([error])`
> - error `<Error>` Optional, an error to emit with `'error'` event.


> https://nodejs.org/api/stream.html#stream_event_error
> 
> After `'error'`, no further events other than `'close'` should be emitted (including `'error'` events).


---

Using just `.autoDestroy = true` should be fine.

> https://nodejs.org/api/stream.html#stream_event_error
> 
> The stream is not closed when the 'error' event is emitted unless the autoDestroy option was set to true when creating the stream.

Reply:
:+1: thanks